libparsec/crates/client/src/invite/claimer.rs

```diff
@@ -304,31 +304,36 @@ struct BaseClaimInitialCtx {
 
 impl BaseClaimInitialCtx {
     async fn do_wait_peer(self) -> Result<BaseClaimInProgress1Ctx, ClaimInProgressError> {
-        // Wait for the other peer
-        let mut result = self._do_wait_peer().await;
-        // If the attempt was automatically cancelled by the other peer, try again once.
-        // Previous attempts are automatically cancelled when a new start greeting attempt is made.
-        // This way, the peers can synchronize themselves more easily during the wait-peer phase,
-        // without requiring the front-end to deal with it.
-        if let Err(ClaimInProgressError::GreetingAttemptCancelled {
-            origin: GreeterOrClaimer::Greeter,
-            reason: CancelledGreetingAttemptReason::AutomaticallyCancelled,
-            ..
-        }) = result
-        {
-            result = self._do_wait_peer().await
+        // Loop over wait peer attempts
+        for attempt in 0.. {
+            let result = self._do_wait_peer().await;
+            // If the attempt was automatically cancelled by the other peer, try again (at most 8 times).
+            // Previous attempts are automatically cancelled when a new start greeting attempt is made.
+            // This way, the peers can synchronize themselves more easily during the wait-peer phase,
+            // without requiring the front-end to deal with it.
+            if let Err(ClaimInProgressError::GreetingAttemptCancelled {
```

Comment:
The `if let Err(..)` followed by a `result?;` is hard to read (see @FirelightFlagboy comment)

You should instead a match:
```rust
match result {
  Ok(()) => (),
  // If the attempt was automatically cancelled by the other peer, try again (at most 8 times).
  // Previous attempts are automatically cancelled when a new start greeting attempt is made.
  // This way, the peers can synchronize themselves more easily during the wait-peer phase,
  // without requiring the front-end to deal with it.
  Err(ClaimInProgressError::GreetingAttemptCancelled { origin, reason, ..})  if attempt < WAIT_PEER_MAX_ATTEMPTS => continue,
  Err(err) => return err,
}
```

Reply:
0e4e548241110f76284bc5f7fb82038f05eb1d43